packages/trace-viewer/src/ui/uiModeTraceView.tsx

```diff
@@ -73,9 +77,34 @@ export const TraceView: React.FC<{
     pollTimer.current = setTimeout(async () => {
       try {
         const model = await loadSingleTraceFile(traceLocation);
-        setModel({ model, isLive: true });
-      } catch {
-        setModel(undefined);
+        setTrace({ type: 'success', model, isLive: true });
+      } catch (e) {
+        const errors = result.errors.flatMap(error => !!error.message ? [error.message] : []);
```

Comment:
So I wonder why something like this instead of this change does not work:
```js
const errors = result.errors.flatMap(error => !!error.message ? [error.message] : []);
const modelWithErrors = new MultiTraceModel([]);
modelWithErrors.errorDescriptors.push(...errors.map(message => ({ message })));
setModel({ model: modelWithErrors, isLive: false });
```

Reply:
I have updated the PR to have that form. By doing so, you lose:
1. Expressiveness in having a discriminated union enumerate possible states (no data vs we have data (live or not) vs we have failed to get data).
2. `model` stability in the error case; it will create a new error model every 500ms and React will do a full rerender of the child tree.
3. A more complete failure error message next to the actions (which you probably didn't like anyway).